StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.cxx

```diff
@@ -240,14 +241,15 @@ Int_t StFcsPi0FinderForEcal::Make() {
          //                        cout<<"trig ID:"<<trigID<<endl;
          if (trigID == 860009 || trigID == 860000) HaveTrig = 1;
       }
-
       if (HaveTrig == 0) return kStOK;
 
       //TOF mult
       StTriggerData* trg = event->triggerData();
       int tofMult = trg->tofMultiplicity();
       //                cout<<"TOF mult="<<tofMult<<endl;
       if (tofMult > 20) return kStOK;
+      */
```

Comment:
Just remove these lines?

Reply:
Removed TriggerId part. This should be done outside this analysis maker, for example in macro using StSpinPool/StTriggerFilterMaker.

TOFmult part remains since it is in correct place and should work, just that it is not working currently when reading from Mudst. I would like to add something like following to StFcsRawHitMaker::readMuDst() (or StMuDst::createStEvent() function), but I cannot make it work. In short,
  const StTriggerData* StMuEvent::triggerData()
cannot feed 
  StEvent::setTriggerData(StTriggerata*) 
due to StStrArray::TestBit(kBelongs) fails, even after removing const by "const_cast". Please let me know if someones mater than me knows solution (without copying entire TriggerData contents!)

if(mEvent->triggerData()==0){                                                                                                                                                                                
        StMuEvent* muevent = mudst->event();                                                                                                                                                                     
        if(muevent){                                                                                                                                                                                             
            StTriggerData* trg = const_cast<StTriggerData*>(muevent->triggerData());                                                                                                                             
            if(trg){                                                                                                                                                                                             
                printf("AAA TOF=%d\n",trg->tofMultiplicity());                                                                                                                                                   
                mEvent->setTriggerData(trg);                                                                                                                                                                     
                // StStrArray::push_back(const TObject*): Assertion `!obj->TestBit(kBelongs)&& obj->TestBit(TObject::kNotDeleted)' failed.                                                                       
                printf("BBB TOF=%d\n",mEvent->triggerData()->tofMultiplicity());                                                                                                                                 
            }                                                                                                                                                                                                    
        }                                                                                                                                                                                                        
    }

Reply:
Can you comment out Line 472: h2_EcalMult_vs_TofMult->Fill(tofMult, n_Ecal_cut);  in StFcsPi0FinderForEcal.cxx . This line also use tofMult which is removed out by Akio. So please remove this line as well.

Reply:
I think we should keep it for now. When reading StTriggerData from Mudst is there (see above), we can put it back. O the other hand, this was too avoid mid-central events from AA collisions, and we may not need it for pp. Still once we have tofmult, we should  study.

Reply:
@akioogawa I'm not sure that this is a valid use of `const_cast` you show here. I would start coding this in a similar way to how it was done in 9e95896b5d07735d5bb90b6b6bca0ea249dabeec, we can see if it can be refactored then.

Reply:
Another attempt following StEventMaker example : 
<pre>
    if(mEvent->triggerData()==0){
        StMuEvent* muevent = mudst->event();
        if(muevent){
            StTriggerData* trg = const_cast<StTriggerData*>(muevent->triggerData());
            if(trg){
                printf("AAA TOF=%d\n",trg->tofMultiplicity());
                AddData(new TObjectSet("StTriggerData",trg));
                TObjectSet *os = (TObjectSet*)GetDataSet("StTriggerData");
                printf("BBB\n");
                if(os){
                    printf("CCC os->IsOwner()=%d\n",os->IsOwner());
                    StTriggerData* pTrg = (StTriggerData*)os->GetObject();
                    if(os->IsOwner() && pTrg){
                        printf("DDD TOF=%d\n",pTrg->tofMultiplicity());
                        os->DoOwner(0);
			printf("EEE\n");
                        mEvent->setTriggerData(trg);
                        printf("FFF TOF=%d\n",mEvent->triggerData()->tofMultiplicity());
                    }
		}
            }
        }
    }

==>

AAA TOF=17
BBB
CCC os->IsOwner()=1
DDD TOF=17
EEE
root4star: .sl73_gcc485/obj/StRoot/St_base/StArray.cxx:276: void StStrArray::push_back(const TObject*): Assertion `!obj->TestBit(kBelongs) && obj->TestBit(TObject::kNotDeleted)' failed.

</pre>